src/layout.rs

```diff
@@ -0,0 +1,124 @@
+pub mod layout_thread;
+
+use crate::{
+    compositing::{BuiltDisplayList, CompositorDisplayListInfo, IpcSender, ScriptToCompositorMsg},
+    html5::node::Node,
+};
+use log::warn;
+use std::sync::mpsc::channel;
+
+// See https://github.com/servo/servo/blob/5d7ed76b79de359ef1de2bdee83b32bd497d7cd8/components/config/opts.rs#L149
+#[derive(Default)]
+pub struct DebugOptions;
+
+// See https://github.com/servo/servo/blob/5d7ed76b79de359ef1de2bdee83b32bd497d7cd8/components/shared/script/lib.rs#L1150
+struct WebrenderIpcSender(IpcSender);
+
+impl WebrenderIpcSender {
+    fn new() -> Self {
+        Self(IpcSender)
+    }
+
+    fn send_display_list(
+        &self,
+        display_list_info: CompositorDisplayListInfo,
+        list: BuiltDisplayList,
+    ) {
+        let (display_list_data, display_list_descriptor) = list.into_data();
+        // TODO: Set up channel somewhere else
+        let (tx, _rx) = channel();
```

Comment:
The Servo code uses IPC.  I've used an mspc channel as a placeholder for whatever would eventually be used.

Reply:
I guess it's not possible, but can you somehow send a pointer over IPC? I think, it's only text. And i also think it's not quite possible to just send a "text-pointer" and interpret it as a pointer on the other side because then the OS goes probably in between

Reply:
I'm thinking what is serizalized as bytes and sent over IPC is something like
```
rectangle [x0, y0, x1, y1]
rectangle [x0, y0, x1, y1]
circle [x, y, radius]
```

But that's just a wild guess.  But I think it's something low-level like that.